### PR TITLE
Add school frontend CRUD, includes create school, edit school, and delete school(with no issue of invalid characters)

### LINKED ISSUE #1: EPIC: Frontend for Schools CRUD (some of which can be done in parallel)
# User Story

* As an admin
* I can do Create/Read/Update/Destroy operations on schools
* So that I can manage the Schools

### Discussion

The backend CRUD for the schools table is already implemented;  you can find the code in the controllers, and test them on either localhost or swagger.  However, as of right now, the frontend is not there.  

Your job is to implement a frontend as in team03 for the schools table.

### Acceptance Criteria

- [ ] There is a menu option for an admin to get to the schools index page where all of the schools are listed
- [ ] There is an option on that page to create a new school
- [ ] There is an edit button for schools that takes me to school edit page where I can edit schools
- [ ] There is an delete button for schools that deletes a school

### Implementation Todos

**F1: Frontend School Implementation**

F1.1: Building create button
- [ ] Make the "create school" button work.
- [ ] Link the create school button to correct API and make a SchoolCreate.js page for that.
- [ ] Make sure it's actually creating a school

F1.2: Building Edit button
- [ ] Link the edit school button to correct API and make a SchoolEdit.js page for that.
- [ ] Make sure it actually allows you to update a school

F1.3: Building Delete button
- [ ] Delete button appears
- [ ] Link the Delete school button to correct API.
- [ ] Make sure it's actually allows you to delete a school

**B1: Bugs in Backend:**

B1.1: Update API
- [ ] You should not be able to update to the school abbrev if it used as the primary key.  If it is not used as the primary key perhaps it should be (similar to UCSBOrganizations in team03).

### PR DESCRIPTION
**Pull request by JUstISAGE**
In this pull request, I fix the issue that character "[" and "]" will cause an error when creating a new school. I also adjust the delete school and edit school so that they match with the changes I did in the backend. 

dokku: http://proj-school-daniel-dev.dokku-09.cs.ucsb.edu

Closes https://github.com/ucsb-cs156-s24/proj-organic-s24-5pm-1/issues/1

### CODE REVIEW

**Review by crosscountrycoder — COMMENTED** [review]
Looks good to me.

File: frontend/src/main/pages/SchoolIndexPage.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from 'react'
 import { useBackend } from 'main/utils/useBackend';
+//import { useNavigate } from 'react-router-dom';
```
Comment (pconrad) [inline frontend/src/main/pages/SchoolIndexPage.js]: Please remove this commented out line

**Review by pconrad — CHANGES_REQUESTED** [review]
commented out code

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**